src/main/java/de/unistuttgart/crosswordbackend/crosswordchecker/CrosswordGenerator.java

```diff
@@ -200,9 +207,7 @@ private ArrayList<Intersection> getIntersections(String word, Crossword crosswor
       }
     }
     intersections.forEach(currentIntersection -> {
-      if (checkIntesection(word, currentIntersection, crossword)) {
-        intersections.remove(currentIntersection);
-      }
+      intersections.removeIf(intersection -> checkIntersection(word, currentIntersection, crossword));
```

Comment:
Why the `forEach`?
That won't work. Simply delete it.
You're getting a `ConcurrentModificationException` otherwise.
The `removeIf` suffices.
And `currentIntersection` should then be renamed to `intersection`.